### PR TITLE
Several Updates with latest info

### PR DESCRIPTION
**Pull request by tom24d**
There are several missing parts with upstream. This PR includes:

- enable KRShaped=true and add&delete its corresponding bits
- add check suite for apis.Validatable and apis.Defaultable
- update DEVELOPMENT.md
- remove bits which does not seem to be used

Basically, shameless copy from knative/sample-controller#291 &  knative/sample-controller#293 

/cc @whaught @aliok
/assign @n3wscott

### CODE REVIEW

**Comment by whaught** [comment]
This looks good! Hopefully after https://github.com/knative/pkg/pull/1430 folks will be even more strongly nudged toward krshaped as it will become the default.

/lgtm

File: pkg/apis/samples/v1alpha1/samplesource_types.go [hunk 1]
```diff
@@ -45,21 +44,24 @@ type SampleSource struct {
...
 }
 
-// Check that SampleSource is a runtime.Object.
-var _ runtime.Object = (*SampleSource)(nil)
-
-// Check that SampleSource can be validated and can be defaulted.
-var _ resourcesemantics.GenericCRD = (*SampleSource)(nil)
-
-// Check that we can create OwnerReferences to a SampleSource.
-var _ kmeta.OwnerRefable = (*SampleSource)(nil)
```
Comment (lberk) [inline pkg/apis/samples/v1alpha1/samplesource_types.go:59]: Why did we drop the `kmeta.OwnerRefable` comment?

Comment (tom24d) [reply pkg/apis/samples/v1alpha1/samplesource_types.go:59]: Oh sorry I am completely forget to leave that. plz see next commit.

File: pkg/apis/samples/v1alpha1/samplesource_types.go [hunk 2]
```diff
@@ -45,21 +44,25 @@ type SampleSource struct {
...
-
-// Check that SampleSource implements the Conditions duck type.
-var _ = duck.VerifyType(&SampleSource{}, &duckv1.Conditions{})
+var (
+	// Check that SampleSource can be validated and defaulted.
+	_ apis.Validatable   = (*SampleSource)(nil)
+	_ apis.Defaultable   = (*SampleSource)(nil)
```
Comment (mattmoor) [inline pkg/apis/samples/v1alpha1/samplesource_types.go:54]: Format Go code:
```suggestion
	_ apis.Validatable = (*SampleSource)(nil)
	_ apis.Defaultable = (*SampleSource)(nil)
```

**Comment by lberk** [comment]
/lgtm
/approve
Thanks!

**Pull request merged**